ral-facilities/datagateway
# Feature/dataset landing page #473

## Description
Adds landing pages (aka detailed views) when in ISIS mode for single datasets, investigations and studies (if browsing the study hierarchy). These aim to replicate existing DOI landing pages, containing information for citation, STFC/ISIS branding, and for studies embedded JSON+LD for Google dataset search.

These use MUI so should be visually consistent with the rest of the app, for example using rounded papers and displaying "short" information like dates on the right as we do in card view. Actions for child entities are also displayed on the right, and viewing all children appears as a tab below the branding header. Fields that are not required will not appear in the UI if we don't get values for them.

Not all of the links from the exising landing pages (https://data.isis.stfc.ac.uk/doi/STUDY/103198140/) have included as some point to Topcat which we'd presumably replace with SciGateway when in production. The layout might also need to be tweaked for real data, if for example we have many child entities or particularly long abstracts.

Changes made and affected files (haven't included `.test` and `.snapshot` files):
- MINOR CHANGE: Added fields to entities so we can display more information about them, and changed `FULL_NAME` to `FULLNAME` (which is how it appears in the schema) allowing us to actually get users' names from our API requests
  - `datagateway-common/src/app.types.tsx`
  - `datagateway-dataview/cypress/integration/card/dls/visits.spec.ts`
  - `datagateway-dataview/cypress/integration/card/isis/instruments.spec.ts`
  - `datagateway-dataview/cypress/integration/card/isis/investigations.spec.ts`
  - `datagateway-dataview/cypress/integration/table/dls/myData.spec.ts`
  - `datagateway-dataview/cypress/integration/table/dls/visits.spec.ts`
  - `datagateway-dataview/cypress/integration/table/isis/instruments.spec.ts`
  - `datagateway-dataview/cypress/integration/table/isis/investigations.spec.ts`
  - `datagateway-dataview/cypress/integration/table/isis/myData.spec.ts`
  - `datagateway-dataview/src/views/detailsPanels/dls/visitDetailsPanel.component.tsx`
  - `datagateway-dataview/src/views/detailsPanels/isis/instrumentDetailsPanel.component.tsx`
  - `datagateway-dataview/src/views/detailsPanels/isis/investigationDetailsPanel.component.tsx`

- MINOR ADDITION: Added STFC logo and string in state to enable hosting it (the host of the plugin can't be easily got when it is served through SG, so we use actions to load it from the settings file into state instead). The logo is used in the new branding headers on landing pages.
  - `datagateway-common/src/images/stfc-logo-white-text.png`
  - `datagateway-dataview/src/state/actions/actions.types.tsx`
  - `datagateway-dataview/src/state/actions/index.tsx`
  - `datagateway-dataview/src/state/app.types.tsx`
  - `datagateway-dataview/src/state/reducers/dgdataview.reducer.tsx`

- MINOR CHANGE: Allowed for include filter when fetching ISIS investigations. Need this so we can get more detailed metadata on landing pages.
  - `datagateway-common/src/state/actions/investigations.tsx`

- MINOR CHANGE: Table/card now link to landing pages (where appropriate) and have a "dummy" tab in the detail panel for viewing all child entities. Previously the main link went directly to a table/card view of all children.
  - `datagateway-dataview/cypress/integration/card/isis/datasets.spec.ts`
  - `datagateway-dataview/cypress/integration/card/isis/investigations.spec.ts`
  - `datagateway-dataview/cypress/integration/card/isis/studies.spec.ts`
  - `datagateway-dataview/cypress/integration/table/isis/datasets.spec.ts`
  - `datagateway-dataview/cypress/integration/table/isis/investigations.spec.ts`
  - `datagateway-dataview/cypress/integration/table/isis/myData.spec.ts`
  - `datagateway-dataview/cypress/integration/table/isis/studies.spec.ts`
  - `datagateway-dataview/src/views/card/isis/isisDatasetsCardView.component.tsx`
  - `datagateway-dataview/src/views/card/isis/isisInvestigationsCardView.component.tsx`
  - `datagateway-dataview/src/views/card/isis/isisStudiesCardView.component.tsx`
  - `datagateway-dataview/src/views/detailsPanels/isis/datasetDetailsPanel.component.tsx`
  - `datagateway-dataview/src/views/detailsPanels/isis/investigationDetailsPanel.component.tsx`
  - `datagateway-dataview/src/views/table/isis/isisDatasetsTable.component.tsx`
  - `datagateway-dataview/src/views/table/isis/isisInvestigationsTable.component.tsx`
  - `datagateway-dataview/src/views/table/isis/isisMyDataTable.component.tsx`
  - `datagateway-dataview/src/views/table/isis/isisStudiesTable.component.tsx`

- MAJOR ADDITION: Addition of the landing pages themselves, along with the branding component.
  - `datagateway-dataview/cypress/integration/landing/isis/dataset.spec.ts`
  - `datagateway-dataview/cypress/integration/landing/isis/investigation.spec.ts`
  - `datagateway-dataview/cypress/integration/landing/isis/study.spec.ts`
  - `datagateway-dataview/src/views/landing/isis/isisBranding.component.tsx`
  - `datagateway-dataview/src/views/landing/isis/isisDatasetLanding.component.tsx`
  - `datagateway-dataview/src/views/landing/isis/isisInvestigationLanding.component.tsx`
  - `datagateway-dataview/src/views/landing/isis/isisStudyLanding.component.tsx `

- MINOR ADDITION: Added strings for new fields and static landing page information (e.g. citation guidlines) which are the same for all entities.
  - `datagateway-dataview/public/res/default.json`

- MINOR CHANGE: Updated routing with landing pages.
  - `datagateway-dataview/src/page/pageContainer.component.tsx`
  - `datagateway-dataview/src/page/pageRouting.component.tsx`

- MEDIUM ADDITION: Created components for action buttons (these are based of the ones we had for card view, but have not yet refactored them out of the card view files, see #461).
  - `datagateway-dataview/src/views/addToCartButton.component.tsx`
  - `datagateway-dataview/src/views/downloadButton.component.tsx`

- MINOR CHANGE: Ignore linting rule for Cypress support commands.
  - `datagateway-search/cypress/support/index.d.ts`
  - `datagateway-dataview/cypress/support/index.d.ts`

## Testing instructions

To get the STFC logo to display, will need to make sure `plugin_host` is set correctly in the settings file (e.g. `http://localhost:3000` for standalone or `http://localhost:5001` for serving through SG)

- [x] Check that when navigating through the ISIS hierarchy the main links for studies, investigations and datasets go to landing pages
- [x] Check that when navigating through the ISIS hierarchy the detail panel tab links for investigations and datasets go to the next table/card view
- [x] Check that the "VIEW <ENTITIES>" tab on landing pages goes to the next table/card view
- [x] Check that the link for a specific child on the landing pages goes to the landing page for that child
- [x] Check that on study landing pages, inspecting the html shows JSON+LD embedded in the pages header
- [x] Check that on the landing pages, everything displays as expected (not that fields we don't populate in the pre-prod data like STUDY.PID won't display)

## Agile board tracking
connect to #473
